libs/libc/stream/lib_libbsprintf.c

```diff
@@ -169,19 +170,17 @@ int lib_bsprintf(FAR struct lib_outstream_s *s, FAR const IPTR char *fmt,
         }
       else if (c == 's')
         {
-          FAR const char *value = data + offset;
-
           if (prec != NULL)
             {
               offset += strtol(prec, NULL, 10);
               prec = NULL;
             }
           else
             {
-              offset += strlen(value) + 1;
+              offset += strlen(var->s) + 1;
             }
 
-          ret += lib_sprintf(s, fmtstr, value);
+          ret += lib_sprintf(s, fmtstr, var->s);
```

Comment:
This seems okay since `var = (FAR void *)(data + offset);` is calculated already on line 89 :-)
Why temporary `const char *` was used here maybe some compilers complained? But `fmtstr` also should be `const char *` while its not :-)

Reply:
it may be mistake from the origin author.